six/modules/c++/cphd/include/cphd/CPHDXMLControl.h

```diff
@@ -45,24 +37,31 @@ namespace cphd
  *  \class CPHDXMLControl
  *
  *  \brief This class converts a Metadata object into a CPHD XML
- *  Document Object Model (DOM).
+ *  Document Object Model (DOM) and vice-versa.
  */
-class CPHDXMLControl : public six::XMLParser
+class CPHDXMLControl
 {
 public:
     /*!
      *  \func CPHDXMLControl
-     *
      *  \brief Default constructor
+     *
+     *  \param log provide logger object
+     *  \param ownLog flag indicates if logging will be taken care of by user
      */
-    CPHDXMLControl();
+    CPHDXMLControl(logging::Logger* log = NULL, bool ownLog = false);
 
-    /*!
-     *  \func CPHDXMLControl
+    //! Destructor
+    virtual ~CPHDXMLControl();
+
+    /*
+     *  \func setLogger
+     *  \brief Handles setting logger based on log provided and ownLog flag
      *
-     *  \brief Constructor with custom log option
+     *  \param log provide logger object
+     *  \param ownLog flag indicates if logging will be taken care of by user
```

Comment:
The flag is for whether the object is responsible for deleting the logger on destruction, isn't it?

Reply:
Yeah that's right. I took a wild guess based on the context it was being used in ...